features/admin.users.v1/components/AccountStatusFilterDropdown.tsx

```diff
@@ -0,0 +1,131 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SelectChangeEvent } from "@mui/material/Select";
+
+import Badge from "@oxygen-ui/react/Badge";
+import Box from "@oxygen-ui/react/Box";
+import Checkbox from "@oxygen-ui/react/Checkbox";
+import FormControl from "@oxygen-ui/react/FormControl";
+import ListItemText from "@oxygen-ui/react/ListItemText";
+import MenuItem from "@oxygen-ui/react/MenuItem";
+import Select from "@oxygen-ui/react/Select";
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import React, { FunctionComponent, ReactElement } from "react";
+
+import { USER_ACCOUNT_STATUS_FILTER_OPTIONS } from "../constants/user-management-constants";
+import { AccountStatusFilterOption } from "../models/user";
+
+import "./AccountStatusFilterDropdown.scss";
+
+/**
+ * Props for the AccountStatusFilterDropdown component.
+ */
+interface AccountStatusFilterDropdownProps extends IdentifiableComponentInterface {
+    selectedFilters: string[];
+    onSelectedFiltersChange: (selected: string[]) => void;
+    label?: string;
+}
+
+/**
+ * The AccountStatusFilterDropdown component.
+ *
+ * @param props - Props injected into the component.
+ * @returns React Element.
+ */
+const AccountStatusFilterDropdown: FunctionComponent<AccountStatusFilterDropdownProps> = ({
+    selectedFilters,
+    onSelectedFiltersChange,
+    label = "Account Status",
+    ["data-componentid"]: componentId = "account-status-filter"
+}: AccountStatusFilterDropdownProps): ReactElement => {
+
+    const handleChange = (event: SelectChangeEvent<string[]>) => {
+        const value: string[] | string = event.target.value;
+
+        onSelectedFiltersChange(typeof value === "string" ? value.split(",") : value);
+    };
+
+    /**
+     * Renders the dropdown label with selection count badge.
+     * @returns The rendered component
+     */
+    const renderLabelWithCount = (): ReactElement => (
+        <Box 
+            className="dropdown-label-container"
+            data-componentid={ `${componentId}-label` }
+        >
+            { label }
+            { selectedFilters.length > 0 && (
+                <Badge
+                    className="status-count-badge"
+                    badgeContent={ selectedFilters.length }
+                    data-testid={ `${componentId}-label-badge` }
+                />
+            ) }
+        </Box>
+    );
+
+    return (
+        <Box 
+            className="account-status-filter-dropdown"
+            data-componentid={ componentId }
+        >
+            <FormControl fullWidth>
+                <Select
+                    className="account-status-select"
+                    id="account-status-select"
```

Comment:
we don't need id, do we?

Reply:
This is not required. Will remove it.

Reply:
Resolved via https://github.com/wso2/identity-apps/pull/8171/commits/59b5a133b5cce5da26541026a8316a7cc32b6305